src/Compiler/Microsoft.CodeAnalysis.Razor.Compiler/src/Language/Components/ComponentDocumentClassifierPass.cs

```diff
@@ -99,7 +99,7 @@ protected override void OnDocumentStructureCreated(
         {
             // We don't want component imports to be considered as real component.
             // But we still want to generate code for it so we can get diagnostics.
-            @class.BaseType = new BaseTypeWithModel("object");
```

Comment:
The original "object" version is actually correct. It was a weird merge back onto 17.12 where both me and Jan had edited the line, but only my change got back ported.